geopandas_view/view.py

```diff
@@ -478,24 +496,142 @@ def _choropleth(
         bins = binning.bins.tolist()
         bins.insert(0, gdf[column].min())
 
-    choro = folium.Choropleth(
-        gdf.__geo_interface__,
-        data=gdf[["__folium_key", column]],
-        key_on="feature.properties.__folium_key",
-        columns=["__folium_key", column],
-        legend_name=column,
-        fill_color=cmap,
-        bins=bins,
-        name=column,
-        **kwds,
-    )
+    geo_data = gdf.__geo_interface__
+    data = gdf[columns]
+
+    # Create color_data dict
+    if hasattr(data, "set_index"):
+        # This is a pd.DataFrame
```

Comment:
```suggestion
    if isinstance(data, pd.DataFrame):
```

I know that your solution works as well, but having a more explicit version feels better.